rusk/src/lib/services/state.rs

```diff
@@ -329,14 +326,31 @@ impl State for Rusk {
         info!("Received Revert request");
 
         let mut state = self.state()?;
-
         state.revert();
-        self.persist(&mut state)?;
 
         let state_root = state.root().to_vec();
         Ok(Response::new(RevertResponse { state_root }))
     }
 
+    async fn persist(
+        &self,
+        request: Request<PersistRequest>,
+    ) -> Result<Response<PersistResponse>, Status> {
+        info!("Received Persist request");
+
+        let request = request.into_inner();
+
+        let mut state = self.state()?;
+
+        if request.state_root == state.root() {
+            return Err(Status::invalid_argument("state root mismatch"));
```

Comment:
Can you add the rusk state root in the error? (Useful to debug imho)

Reply:
Ah yes, will do!